compiler/noirc_driver/src/lib.rs

```diff
@@ -121,8 +122,13 @@ pub fn check_crate(
     crate_id: CrateId,
     deny_warnings: bool,
 ) -> CompilationResult<()> {
+    #[cfg(not(feature = "aztec"))]
+    let macros: Vec<&dyn MacroProcessor> = Vec::new();
+    #[cfg(feature = "aztec")]
+    let macros = vec![&aztec_macros::AztecMacro as &dyn MacroProcessor];
```

Comment:
We _could_ make this a runtime flag in nargo, ie users will need to do `nargo compile --macros aztec` to compile with the aztec macro

Reply:
This would atleast make it so that we don't have two binaries so the lsp can be configured to run it by just modifying the settings.json and still isolate this aztec feature

Reply:
#3596